### PR TITLE
Add action to open a subtree recursively (v2)

### PR DESCRIPTION
**Pull request by lambdalisue**
Original PR is #480 

I rewrote `fern#internal#node#expand_tree` because of #482

### CODE REVIEW

**Comment by brandon1024** [comment]
Good first revision! Recursive tree expansion works as I'd expect, very nice :-)

I found a small quirk though. I have some configuration that toggles expansion of a node or opens a leaf. This has stopped working correctly with this revision.

```vim
	nmap <buffer> <CR> <Plug>(fern-action-toggle-expand-open)

	nmap <buffer><expr> <Plug>(fern-action-toggle-expand-open)
		\ fern#smart#leaf(
		\   "<Plug>(fern-action-open:select)",
		\   "<Plug>(fern-action-expand)",
		\   "<Plug>(fern-action-collapse)")
```

When I open Fern and expand a tree for the first time (with `<CR>`), it works as I would expect. After collapsing the node and expanding the tree again, the first child node is in the `expanded` state but doesn't render as such.

Here's a screen capture to demonstrate this behaviour. Notice the state of `ews-eks` when I expand it the second time:

Peek 2023-08-26 08-04

File: doc/fern.txt [hunk 1]
```diff
@@ -909,6 +909,21 @@ GLOBAL							*fern-mapping-global*
...
+	Recursively expand the tree on a cursor node and keep the cursor on the 
+	root node (moves to the first child node after the cursor node)
+
+*<Plug>(fern-action-expand-tree)*
+	An alias to "expand-tree:in" action. Users can overwrite this mapping to
+	change the default behavior of "expand-tree" action like:
+>
+	nmap <buffer>
+	      \ <Plug>(fern-action-expand-tree)
+	      \ <Plug>(fern-action-expand-tree:stay)
```
Comment (brandon1024) [inline doc/fern.txt:927]: I think we are missing the closing quote block character here (`<`).

File: autoload/fern/helper/async.vim [hunk 2]
```diff
@@ -156,6 +156,31 @@ function! s:async_expand_node(key) abort dict
...
+        \   fern.nodes,
+        \   fern.provider,
+        \   fern.comparator,
+        \   fern.source.token,
+        \ )
+        \})
+        \.then({ ns -> self.update_nodes(ns) })
+        \.finally({ -> Profile() })
+endfunction
+let s:async.expand_tree = funcref('s:async_expand_tree')
```
Comment (brandon1024) [inline autoload/fern/helper/async.vim:182]: Is this something that we should document in `fern-develop.txt`?

Comment (LunarWatcher) [reply autoload/fern/helper/async.vim:182]: Probably should. The other async functions are part of the dev documentation IIRC

**Comment by brandon1024** [comment]
@lambdalisue @LunarWatcher does it make sense to close one of the PRs we have open? Who wants to own the changes for this feature?

**Comment by lambdalisue** [comment]
@brandon1024 @LunarWatcher fixed and force pushed. Could you try and check?

**Comment by LunarWatcher** [comment]
LGTM

**Pull request merged**